src/debug.c

```diff
@@ -962,14 +964,25 @@ NULL
     {
         server.pause_cron = atoi(c->argv[2]->ptr);
         addReply(c,shared.ok);
-    } else if (!strcasecmp(c->argv[1]->ptr,"replybuffer-peak-reset-time") && c->argc == 3 ) {
-        if (!strcasecmp(c->argv[2]->ptr, "never")) {
-            server.reply_buffer_peak_reset_time = -1;
-        } else if(!strcasecmp(c->argv[2]->ptr, "reset")) {
-            server.reply_buffer_peak_reset_time = REPLY_BUFFER_DEFAULT_PEAK_RESET_TIME;
-        } else {
-            if (getLongFromObjectOrReply(c, c->argv[2], &server.reply_buffer_peak_reset_time, NULL) != C_OK)
+    } else if (!strcasecmp(c->argv[1]->ptr,"replybuffer") && c->argc == 4 ) {
+        if(!strcasecmp(c->argv[2]->ptr, "peak-reset-time")) {
+            if (!strcasecmp(c->argv[3]->ptr, "never")) {
+                server.reply_buffer_peak_reset_time = -1;
+            } else if(!strcasecmp(c->argv[3]->ptr, "reset")) {
+                server.reply_buffer_peak_reset_time = REPLY_BUFFER_DEFAULT_PEAK_RESET_TIME;
+            } else {
+                if (getLongFromObjectOrReply(c, c->argv[3], &server.reply_buffer_peak_reset_time, NULL) != C_OK)
+                    return;
+            }
+        } else if(!strcasecmp(c->argv[2]->ptr,"resizing")) {
+            if (!strcasecmp(c->argv[3]->ptr, "on")) {
+                server.reply_buffer_resizing_enabled = 1;
+            } else if(!strcasecmp(c->argv[3]->ptr, "off")) {
+                server.reply_buffer_resizing_enabled = 0;
+            } else {
+                addReplyErrorFormat(c, "Argument must be either 'on' or 'off'");
```

Comment:
maybe easier to just use a generic syntax error, i don't think we need to be that explicit here.
also, maybe we can just convert the input into a "boolean" using atoi (like done in `set-active-expire`).

p.e. maybe add an `else` with syntax error in case it's not one of the two "sub-commands"